src/components/Activity/ActivityHub.js

```diff
@@ -1,42 +1,66 @@
-import React from 'react'
+import React, { useMemo } from 'react'
 import PropTypes from 'prop-types'
-import { Spring, Transition, animated } from 'react-spring'
 import styled from 'styled-components'
-import { theme, IconClose } from '@aragon/ui'
-import TimeTag from './TimeTag'
+import { Transition, animated } from 'react-spring'
+import {
+  IconAttention,
+  IconCheck,
+  IconClose,
+  IconError,
+  SafeLink,
+  blockExplorerUrl,
+  theme,
+} from '@aragon/ui'
 import { activityStatusTypes } from '../../contexts/ActivityContext'
+import { network } from '../../environment'
+import TimeTag from './TimeTag'
+import AppIcon from '../AppIcon/AppIcon'
 
 const spring = { tension: 1900, friction: 200, precision: 0.0001, clamp: true }
 
-function ActivityHub({ activities, keys, clearActivity }) {
+function ActivityHub({
+  activities,
+  keys,
+  clearActivity,
+  getAppByProxyAddress,
+}) {
   const [activitiesReady, setActivitiesReady] = React.useState({})
 
+  const activityItems = useMemo(
+    () => activities.sort((a, b) => b.createdAt - a.createdAt),
+    [activities, getAppByProxyAddress]
```

Comment:
```suggestion
    [activities]
```
Shouldn't merely `activities` suffice?